### ISSUE TITLE
support "out" and "keepdims" arguments where applicable

### ISSUE THREAD

**Issue by gdementen** [kwgoodman/bottleneck#97]
numpy has those on most aggregate functions. Having them would make bottleneck much more of a drop-in replacement than currently.

**Comment by shoyer** [comment]
`keepdims` is pretty easy to implement -- you can just use a `reshape` afterwards to insert dummy dimensions (or even just adjust the strides/shape manually).

**Comment by gdementen** [comment]
I know, but I would like to be able to just do this:

try:
    import bottleneck as bn
    nanmin = bn.nanmin
    ...
except ImportError:
   nanmin = np.nanmin
   ...

However, if the functions have not the same signatures, I can't.

**Comment by shoyer** [comment]
@gdementen Yes, I was suggesting that it would be pretty easy to add `keepdims` to bottleneck functions.

**Comment by kwgoodman** [comment]
Yeah, ``keepdims`` is the easier of the two.

``out`` would be nice because it would make the functions faster.

**Comment by kwgoodman** [comment]
I played around with adding an `out` argument to bn.move_sum.

All moving window functions in bottleneck use the mover function. That function does not know how the caller (here bn.move_sum) maps input dtype to output dtype. Therefore I hacked in a fixed output dtype (float64).

I don't see any difference in speed for (1000, 1000) input. For (4000, 4000) input I got an improvement in runtime from 35ms to 27ms (not counting the time to run out=np.empty((4000, 4000))).

Adding an `out` input argument would take a bit of a redesign. The reason I took a look at adding `out` is that I'm thinking of adding some utilities to bottleneck that break an input array into chunks and work on the chunks in a pool of threads.
